### PR TITLE
StatusEffect and similiar renames as per #1036

### LINKED ISSUE #1036: StatusEffectInstance -> StatusEffect, SE -> SEType, SEType -> SECategory
(`SE` = `StatusEffect`)
This would make it match things like `Entity` and `EntityType`.

### PR DESCRIPTION
**Pull request by dexman545**
StatusEffectType -> StatusEffectCategory (#1036)
StatusEffect -> StatusEffectType
StatusEfectInstance -> StatusEffect
(and related names)
Closes #1036

### CODE REVIEW

File: mappings/net/minecraft/util/registry/Registry.mapping [hunk 1]
```diff
@@ -18,7 +18,7 @@ CLASS net/minecraft/class_2378 net/minecraft/util/registry/Registry
 	FIELD field_11156 SOUND_EVENT Lnet/minecraft/class_2378;
 	FIELD field_11157 CARVER Lnet/minecraft/class_2378;
 	FIELD field_11158 CUSTOM_STAT Lnet/minecraft/class_2378;
-	FIELD field_11159 STATUS_EFFECT Lnet/minecraft/class_2378;
+	FIELD field_11159 STATUS_EFFECT_TYPE Lnet/minecraft/class_2378;
```
Comment (Juuxel) [inline mappings/net/minecraft/util/registry/Registry.mapping]:
```suggestion
	FIELD field_11159 STATUS_EFFECT Lnet/minecraft/class_2378;
```
I think this is fine without the type suffix; its ID is `mob_effect` and we have other `*Type` registries without the suffix (`SCREEN_HANDLER`, `RULE_TEST` iirc).

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]
Content looks good, though I don't agree with the rename

**Comment by liach** [comment]
Mind fix the merge conflicts?

File: mappings/net/minecraft/util/registry/Registry.mapping (hunk 1 above)
Comment (Juuxel) [inline mappings/net/minecraft/util/registry/Registry.mapping]: Huh? I thought you changed it back to `STATUS_EFFECT` in 2924a85...

**Review by Runemoro — APPROVED** [review]

**Review by YanisBft — CHANGES_REQUESTED** [review]
Please no, registry name is `status_effect`, not `status_effect_type`

**Comment by Juuxel** [comment]
There are other `Type`s registered in registries without a `_type` suffix:
- Rule test types in `rule_test`
- Pos rule test types in `pos_rule_test`
- Menu types in `menu`

**Pull request closed without merging**